tools/gfx/render.h

```diff
@@ -277,36 +261,48 @@ class Resource: public Slang::RefObject
         int cpuAccessFlags = 0;     ///< Combination of Resource::AccessFlag
     };
 
-        /// Get the type
-    SLANG_FORCE_INLINE Type getType() const { return m_type; }
-        /// True if it's a texture derived type
-    SLANG_FORCE_INLINE bool isTexture() const { return int(m_type) >= int(Type::Texture1D); }
-        /// True if it's a buffer derived type
-    SLANG_FORCE_INLINE bool isBuffer() const { return m_type == Type::Buffer; }
-
-        /// Get the descBase
-    const DescBase& getDescBase() const;
-        /// Returns true if can bind with flag
-    bool canBind(BindFlag::Enum bindFlag) const { return getDescBase().canBind(bindFlag); }
-
-        /// For a usage gives the required binding flags
-    static const BindFlag::Enum s_requiredBinding[];    /// Maps Usage to bind flags required
-
-    protected:
-    Resource(Type type):
-        m_type(type)
-    {}
-
-    static void compileTimeAsserts();
+    inline static BindFlag::Enum getDefaultBindFlagsFromUsage(IResource::Usage usage)
```

Comment:
It seems like `BindFlag` and `IResource::Usage` are overlapping implementations of the same basic idea. I suspect they were added by two different people who had differing ideas about how all this stuff should work.

Once we get to refactoring, I would like it if we could drop down to just having one or the other.